bin/hog/batchrecv.C

```diff
@@ -150,66 +165,60 @@ void runRecvConnection(SessionGroup* sg, int c, std::string dir) {
 
     auto txnF = appendStorageSession(sg, instantiateDir(group, dir), (storage::PipeQOS)qos, (storage::CommitMethod)cm, stmts);
 
-    ssend(c, &ack, sizeof(ack));
+    connection->send(&ack, sizeof(ack));
 
     // now that we've prepared a log file,
     // just throw everything that we read into it
     while (true) {
-      srecv(c, &inb);
+      receiveIntoBuffer(*connection, &inb);
       gzbuffer zb(inb, &outb);
 
       while (!zb.eof()) {
-        size_t n = 0;
+        uint64_t n = 0;
         read(&zb, &n);
-        txn.reserve(n);
-        read(&zb, txn.data, n);
+        txn.resize(n);
+        read(&zb, txn.data(), txn.size());
 
-        storage::Transaction stxn(txn.data, n);
+        storage::Transaction stxn(txn.data(), txn.size());
         txnF(stxn);
       }
 
-      ssend(c, &ack, sizeof(ack));
+      connection->send(&ack, sizeof(ack));
     }
   } catch (std::exception& ex) {
-    out << "terminating log session with error: " << ex.what() << std::endl;
-    close(c);
+    out() << "terminating log session with error: " << ex.what() << std::endl;
   }
 }
 
-void runRecvServer(int socket, std::string dir, bool consolidate) {
+void runRecvServer(std::unique_ptr<NetServer> server, std::string dir, bool consolidate) {
   SessionGroup* sg = makeSessionGroup(consolidate);
   std::vector<std::thread> cthreads;
 
-  hobbes::registerEventHandler(
-    socket,
-    [sg,&cthreads,dir](int socket) {
-      // accept and make a new thread for this connection
-      // (not the most efficient way but good for a start)
-      int c = accept(socket, 0, 0);
-      if (c < 0) {
-        out << "failed to accept network connection: " << strerror(errno) << std::endl;
-      } else {
-        cthreads.push_back(std::thread(std::bind(&runRecvConnection, sg, c, dir)));
-      }
+  while (true) {
+    auto conn = server->accept();
```

Comment:
does it imply the execution of server doesn't care about other potential events (we don't any at the moment admittedly) ?